plugins/services/src/js/components/forms/GeneralServiceFormSection.js

```diff
@@ -92,7 +92,10 @@ class GeneralServiceFormSection extends Component {
     const initialIsExpanded = this.shouldShowAdvancedOptions();
 
     return (
-      <AdvancedSection initialIsExpanded={initialIsExpanded}>
+      <AdvancedSection
+        initialIsExpanded={initialIsExpanded}
+        shouldExpand={expandAdvancedSettings}
```

Comment:
Why can't we reuse `initialIsExpanded` and need separate `shouldExpand`?

Reply:
Good question. I think it will be confusing to use `initialIsExpanded` for non-initial expansion.

Reply:
Having two props that do the same thing is less confusing?

Reply:
They both expand it, but in different scenarios.

Reply:
How those scenarios are different? Can we converge to one prop that controls whether the section is expanded or not?

Reply:
I will try to do it with one prop. Thank you for the helpful feedback.

Reply:
I think if we want to use `getDerivedStateFromProps` like @mperrotti suggested, we will need two separate props. Otherwise, as soon as we change anything in the form, `initialIsExpanded` becomes true and we can't close the advanced settings.

Reply:
We can't close them until we fix the error, right?

Reply:
You mean that we shouldn't we able to?

Reply:
once we move the expand on error logic here we not going to need the second prop.